src/crawlers/cheerio_crawler.js

```diff
@@ -328,21 +350,28 @@ class CheerioCrawler {
      */
     async _handleRequestFunction({ request, autoscaledPool }) {
         if (this.prepareRequestFunction) await this.prepareRequestFunction({ request });
-        const { dom, responseStream: response } = await addTimeoutToPromise(
+        const { dom, isXmlOrHtml, body, responseStream: response } = await addTimeoutToPromise(
             this._requestFunction({ request }),
             this.requestTimeoutMillis,
             'CheerioCrawler: requestFunction timed out.',
         );
 
         request.loadedUrl = response.url;
 
-        const $ = cheerio.load(dom);
+        const $ = dom ? cheerio.load(dom, { xmlMode: isXmlOrHtml }) : null;
         const context = {
             $,
             // Using a getter here not to break the original API
             // and lazy load the HTML only when needed.
             get html() {
-                return $.html({ decodeEntities: false });
+                log.deprecated('CheerioCrawler: Parameter html for handlePageFunction is deprecated use body instead.');
```

Comment:
CheerioCrawler: The "html" parameter of handlePageFunction is deprecated, use "body" instead.